### DISCUSSION TITLE
About the "max_value_entropy_search.py"

### DISCUSSION THREAD

**Discussion by LiuYang-1997** [meta-pytorch/botorch#969] [Q&A] [answered]
In "_compute_information_gain" method of "max_value_entropy_search.py", what does "mean_M","variance_M" and "covar_mM" mean? Do they represent the values when there is no noise predicted by the model?And I don't know what "X_pending" means.
Someone help me,please,I am just a novice.TAT
Thank you very much.
```
    def _compute_information_gain(
        self, X: Tensor, mean_M: Tensor, variance_M: Tensor, covar_mM: Tensor
    ) -> Tensor:
        r"""Computes the information gain at the design points `X`.

        Approximately computes the information gain at the design points `X`,
        for both MES with noisy observations and multi-fidelity MES with noisy
        observation and trace observations.

        The implementation is inspired from the papers on multi-fidelity MES by
        [Takeno2020mfmves]_. The notation in the comments in this function follows
        the Appendix C of [Takeno2020mfmves]_.

        `num_fantasies = 1` for non-fantasized models.

        Args:
            X: A `batch_shape x 1 x d`-dim Tensor of `batch_shape` t-batches
                with `1` `d`-dim design point each.
            mean_M: A `batch_shape x num_fantasies x (m)`-dim Tensor of means.
            variance_M: A `batch_shape x num_fantasies x (m)`-dim Tensor of variances.
            covar_mM: A
                `batch_shape x num_fantasies x (m) x (1 + num_trace_observations)`-dim
                Tensor of covariances.

        Returns:
            A `num_fantasies x batch_shape`-dim Tensor of information gains at the
            given design points `X` (`num_fantasies=1` for non-fantasized models).
        """



X_pending: `m x d` Tensor with `m` `d`-dim design points that have been submitted for evaluation but have not yet been evaluated.
```

**Comment by LiuYang-1997** [comment]
help me please, TAT

**Comment by liangshi7** [comment, marked as answer]
HI @LiuYang-1997, 

Thanks for asking. The notation is definitely not very human-readable. We basically follow the notation in the paper that is referred to, https://arxiv.org/pdf/1901.08275.pdf (appendix C). In a multi-fidelity setting, _m means values from low fidelities or from the fidelity other than the target fidelity, whereas _M denotes the variables from the highest fidelity (or the target fidelity). In a non-multi-fidelity setting, I think you interpretation is right: basically _m corresponds to the variables from noisy evaluation, whereas _M denotes the variables from the non_noisy evaluation, the function we try to optimize. 

X_pending are the design points that have been submitted to evaluate but their results of them are not returned yet. In this case, we don't want the acquisition function still returns these points to make the sampling more efficient

**Reply by LiuYang-1997** [reply]
Thank you for your answer. Thank you very much.TAT
